src/main/java/com/microsoft/graph/serializer/CalendarSerializer.java

```diff
@@ -51,39 +51,73 @@ private CalendarSerializer() {
      * @throws java.text.ParseException the parse exception
      */
     public static Calendar deserialize(final String strVal) throws ParseException {
-        // Change Z to +0000 to adapt the string to a format
+        // Change Z to adapt the string to a format
         // that can be parsed in Java
         final boolean hasZ = strVal.indexOf('Z') != -1;
+        final boolean hasDot = strVal.indexOf('.') != -1;
+		
+
         String modifiedStrVal;
         final String zSuffix;
-        if (hasZ) {
+        if (hasZ && hasDot) {
+        	zSuffix = "";
+            modifiedStrVal = strVal.replace("Z", "+00:00");
+        } else if (hasZ && !hasDot) {
             zSuffix = "Z";
             modifiedStrVal = strVal.replace("Z", "+0000");
         } else {
             zSuffix = "";
             modifiedStrVal = strVal;
         }
-
-        // Parse the well-formatted date string.
+        
+        final boolean hasOffset = modifiedStrVal.contains("T")
+				? (modifiedStrVal.substring(modifiedStrVal.indexOf('T') + 1).contains("+"))
+						|| (modifiedStrVal.substring(modifiedStrVal.indexOf('T') + 1).contains("-"))
+				: false;
+						
+        // Parse the well-formatted date string with and without offsets (eg: 2019-06-21T17:12:35.138, 2019-06-21T17:12:35.138+0000, 2019-06-21T17:12:35.138-07:00)
         final String datePattern;
         if (modifiedStrVal.contains(".")) {
```

Comment:
hasDot already has if date string contains dot or not. No need to compute again.

Reply:
Done.